Dockerfile

```diff
@@ -1,36 +1,43 @@
-# Builder
-FROM node:10-slim as builder
+# Freebind build
+FROM node:12-slim as freebind
 
 RUN apt-get update \
- && apt-get install -y git python build-essential libicu-dev
+ && apt-get install -y git build-essential
 
-RUN git clone https://github.com/matrix-org/freebindfree.git \
- && cd freebindfree \
- && make
+RUN git clone https://github.com/matrix-org/freebindfree.git
+RUN cd freebindfree && make
 
-COPY ./package.json ./package.json
-RUN npm install
+# Typescript build
+FROM node:12-slim as builder
+
+WORKDIR /build
+
+RUN apt-get update && apt-get install -y git python3 libicu-dev build-essential
+
+COPY ./package.json /build/package.json
+COPY ./package-lock.json /build/package-lock.json
+COPY ./src /build/src
+COPY ./tsconfig.json /build/tsconfig.json
+COPY ./types /build/types
```

Comment:
you can avoid all of these by using a `.dockerignore` file btw (ie: make it a single COPY operation)